helper/error-handler.ts

```diff
@@ -1,80 +1,86 @@
-   * 404  Not Found
-   * 422  Unprocessable Entity
-   * 500  Internal Server Error
-   * 502  Bad Gateway
-   * 504  Gateway Timeout
-   */
-
-  /**
-   * The error message looks like: [404] Not Found. User does not exist
-   */
-  switch (error.statusCode) {
-    case 400:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Bad request'}`;
-    case 401:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Unauthorized'}`;
-    case 403:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Forbidden'}`;
-    case 404:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Not Found'}`;
-    case 409:
-      throw `[400] ${error.name ? `${error.name}. ` : ''}${error.message || 'Conflict'}`;
-    case 422:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Unprocessable Entity'}`;
-    case 429:
-      throw `[400] ${error.name ? `${error.name}. ` : ''}${error.message || 'Too Many Requests'}`;
-    case 500:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Internal Server Error'}`;
-    case 502:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Bad Gateway'}`;
-    case 503:
-      throw `[500] ${error.name ? `${error.name}. ` : ''}${error.message || 'Service Unavailable'}`;
-    case 504:
-      throw `[${error.statusCode}] ${error.name ? `${error.name}. ` : ''}${error.message || 'Gateway Timeout'}`;
-    default:
-      throw `[500] ${error.name ? `${error.name}. ` : ''}${error.message || 'Internal Server Error'}`;
-  }
-}
+import { HttpError } from '@errors/http/http-error';
+import { RuntimeError } from '@errors/runtime/runtime-error';
+import { ErrorStatusCode } from '@helper/app-error';
+import { log } from '@helper/logger';
+import { format } from '@redtea/format-axios-error';
+import { AxiosError } from 'axios';
+
+export function errorHandler(caughtError: Error | HttpError | AxiosError | RuntimeError): undefined {
+  let error = caughtError;
+  const axiosError = (caughtError as AxiosError).isAxiosError && format(caughtError as AxiosError);
+
+  if (!(error instanceof HttpError)) {
+    /**
+     * It means that error was unexpected and can have unpredictable structure
+     * For example, Axios errors have different structure
+     * Also, we can send all unexpected errors to Sentry here
+     */
+    log('This error was not generated by us. We should extract the statusCode, name and message here');
+
+    if (axiosError) {
+      error = {
+        statusCode: axiosError.response?.status as ErrorStatusCode,
+        message: axiosError.response?.statusText || axiosError.message,
+        name: '',
+      };
+
+      if (error instanceof RuntimeError) {
+        error = { statusCode: 400, name: 'Bad request', message: error.message };
+      }

```

Comment:
@khoroshilov-flo What do you think about this? Should we have some kind of converter from RuntimeError to HttpError in order to follow the Single Responsibility Principle? So we won't change errorHandler for any new RuntimeError conversion.